src/stores/models/Oracle.test.js

```diff
@@ -0,0 +1,206 @@
+    const input = observable({
+      txid: '0',
+      amounts: [decimalToSatoshi('99'), decimalToSatoshi('100')],
+      consensusThreshold: decimalToSatoshi('100'),
+      address: undefined,
+      topicAddress: undefined,
+      resultSetEndTime: '10',
+      endTime: '20',
+      options: ['1', '2'],
+      token: QTUM,
+      optionIdxs: [0, 1],
+      status: OracleStatus.CREATED,
+      hashId: 'hahaha',
+    });
+    oracle = new Oracle(input, app);
+
+    expect(oracle.app).toBe(app);
+    expect(oracle.amounts.length).toBe(input.amounts.length);
+    expect(oracle.amounts[0]).toBe(99);
+    expect(oracle.amounts[1]).toBe(100);
+    expect(oracle.consensusThreshold).toBe(100);
+    expect(oracle.url).toBe('/oracle/hahaha');
+    expect(oracle.endTime).toBe('20');
+    expect(oracle.options.length).toBe(input.options.length);
+
+    expect(oracle.isUpcoming).not.toBeTruthy();
+    expect(oracle.unconfirmed).toBeTruthy();
+    expect(oracle.isPending).toBeTruthy();
+    expect(oracle.isArchived).not.toBeTruthy();
+    expect(oracle.isOpenResultSetting).not.toBeTruthy();
+    expect(oracle.phase).toBe(BETTING);
+  });
+
+  it('Constructor Open Result Setting Oracle', async () => {
+    const input = observable({
+      txid: '0',
+      amounts: [decimalToSatoshi('99'), decimalToSatoshi('100')],
+      consensusThreshold: decimalToSatoshi('100'),
+      address: undefined,
+      topicAddress: '4044f951857f2885d66d29a475235dacdaddea84',
+      resultSetEndTime: '10',
+      endTime: '20',
+      options: ['1', '2'],
+      token: QTUM,
+      optionIdxs: [0, 1],
+      status: OracleStatus.OPEN_RESULT_SET,
+      hashId: 'hahaha',
+    });
+    oracle = new Oracle(input, app);
+
+    expect(oracle.app).toBe(app);
+    expect(oracle.amounts.length).toBe(input.amounts.length);
+    expect(oracle.amounts[0]).toBe(99);
+    expect(oracle.amounts[1]).toBe(100);
+    expect(oracle.consensusThreshold).toBe(100);
+    expect(oracle.url).toBe('/oracle/4044f951857f2885d66d29a475235dacdaddea84/undefined/0');
+    expect(oracle.phase).toBe(RESULT_SETTING);
+    expect(oracle.endTime).toBe('10');
+    expect(oracle.options.length).toBe(input.options.length);
+
+    expect(oracle.isUpcoming).not.toBeTruthy();
+    expect(oracle.unconfirmed).not.toBeTruthy();
+    expect(oracle.isPending).not.toBeTruthy();
+    expect(oracle.isArchived).not.toBeTruthy();
+    expect(oracle.isOpenResultSetting).toBeTruthy();
+  });
+
+  it('Archived Voting Oracle', async () => {
+    const input = observable({
+      txid: '0',
+      amounts: [decimalToSatoshi('99'), decimalToSatoshi('100')],
+      consensusThreshold: decimalToSatoshi('100'),
+      address: undefined,
+      topicAddress: '4044f951857f2885d66d29a475235dacdaddea84',
+      resultSetEndTime: '10',
+      endTime: '20',
+      options: ['1', '2'],
+      token: BOT,
+      optionIdxs: [1],
+      status: OracleStatus.WITHDRAW,
+      hashId: 'hahaha',
```

Comment:
same

Reply:
Will be more serious on testings, sorry for being unprofessional.

Thx for reviewing.

Reply:
no problem. its not really about unprofessionalism. its more about "simulating" what actual live data you will be getting. not really a necessity, but my methodology is to re-create real life situations for testing as close as possible.